src/pipx/main.py

```diff
@@ -614,19 +616,68 @@ def get_command_parser() -> argparse.ArgumentParser:
     return parser
 
 
+def setup_file_handler() -> logging.Handler:
+    max_logs = 10
+    # don't use utils.mkdir, to prevent emission of log message
+    constants.PIPX_LOG_DIR.mkdir(parents=True, exist_ok=True)
+
+    existing_logs = sorted(constants.PIPX_LOG_DIR.glob("cmd_*.log"))
+    if len(existing_logs) > max_logs:
+        for existing_log in existing_logs[:-max_logs]:
+            try:
+                existing_log.unlink()
+                print(f"Removed file {existing_log}", file=sys.stderr)
+            except FileNotFoundError:
+                pass
+
+    datetime_str = time.strftime("%Y-%m-%d_%H.%M.%S")
+    log_file = constants.PIPX_LOG_DIR / f"cmd_{datetime_str}.log"
+    counter = 1
+    while log_file.exists():
```

Comment:
Suggest adding a `and counter < max_logs` to be defensive. While loops always scare me.